src/components/stillage/StillageList.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from 'react'
+import { styled } from '@linaria/react'
+import StillageItem from './StillageItem'
+import { useDispatch, useSelector } from 'react-redux'
+import { RootState } from '../../redux/store'
+import { clearStillageList, setStillageList } from '../../redux/slices/stillageSlice'
+import { useParams } from 'react-router-dom'
+import { api } from '../../api'
+
+const StyledList = styled.ul`
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+
+  @media screen and (min-width: 820px) {
+    flex-direction: row;
+    flex-wrap: wrap;
+    li {
+      flex: 0 1 48.5%;
+    }
+  }
+  @media screen and (min-width: 1440px) {
+    gap: 20px;
+  }
+`
+
+function StillageList() {
+  const { id: stillageId } = useParams()
+  const dispatch = useDispatch()
+
+  async function fetchData() {
+    const shelves = await api.shelves({ stillage: stillageId })
+    if (Array.isArray(shelves)) {
+      dispatch(setStillageList(shelves))
+    }
+  }
```

Comment:
If you need to create function inside component body because of internal deps, you should use [`useCallback` hook](https://react.dev/reference/react/useCallback)

Reply:
Thank you for this piece of advice. I've added the useCallback hook